master/buildbot/status/builder.py

```diff
@@ -322,12 +328,27 @@ def getLastFinishedBuild(self):
             b = self.getBuild(-2)
         return b
 
-    def setCategory(self, category):
+    def getTags(self):
+        return self.tags
+
+    def setTags(self, tags):
         # used during reconfig
-        self.category = category
+        self.tags = tags
+
+    def matchesAnyTag(self, tags):
+        return self.tags and any((tag in self.tags) for tag in tags)
```

Comment:
builder.matchesAnyTags(["foo"]) should be False if that builder has no tags (eg, looking for the "Linux" builders should not match things that never opted in to be "Linux" builders).

builder.matchesAnyTags([]) ... I am not sure what this should be. I could make arguments both ways.